packages/dashboard/src/components/queryEditor/queryEditor.tsx

```diff
@@ -18,6 +20,16 @@ export function QueryEditor({
   const [_query, setQuery] = useQuery();
   const addButtonDisabled = useIsAddButtonDisabled(selectedWidgets);
   const correctSelectionMode = getCorrectSelectionMode(selectedWidgets);
+  const { assetModelId, assetIds } = useModelBasedQuery();
+  const { assetSummaries } = useAssetsForAssetModel({
+    assetModelId,
+    iotSiteWiseClient,
+    fetchAll: true,
+  });
+
+  const currentSelectedAsset = assetSummaries.find(
```

Comment:
will assetSummaries ever be undefined?

Reply:
if the api errors out, or there are no models :)

Reply:
no, but in case there is no assets retrieved, currentSelectedAsset will be undefined which has handling in the query editors.

Reply:
unless the query errors out